opportunity_job_app/src/api/baseApi.ts

```diff
@@ -3,30 +3,35 @@
 import axios from "axios";
 import { AuthService } from "./authService";
 
-const api = axios.create({
+const axiosInstance = axios.create({
     baseURL: process.env.NEXT_PUBLIC_API_URL,
 });
 
 const API = {
     getResourceList: (resource: string, queryParams = "") => {
-        return api.get(`${ resource }/?${ queryParams }`);
+        const config = { headers: AuthService.getAuthorizationHeaders() };
+        return axiosInstance.get(`${ resource }/?${ queryParams }`, config);
     },
     getProtectedResourceList: (resource: string, queryParams = "") => {
         const config = { headers: AuthService.getAuthorizationHeaders() };
-        return api.get(`${ resource }/?${ queryParams }`, config);
+        return axiosInstance.get(`${ resource }/?${ queryParams }`, config);
     },
     getResourceDetails: (resource: string, resourceId: number) => {
-        return api.get(`${ resource }/${ resourceId }/`);
+        const config = { headers: AuthService.getAuthorizationHeaders() };
+        return axiosInstance.get(`${ resource }/${ resourceId }/`, config);
     },
     getProtectedResourceDetails: (resource: string, resourceId: number) => {
         const config = { headers: AuthService.getAuthorizationHeaders() };
-        return api.get(`${ resource }/${ resourceId }/`, config);
+        return axiosInstance.get(`${ resource }/${ resourceId }/`, config);
     },
     post: (url: string, data: Object, config: Object) => {
-        return api.post(url, data, config);
+        return axiosInstance.post(url, data, config);
     },
     patch: (url: string, data: Object) => {
-        return api.patch(url, data);
+        return axiosInstance.patch(url, data);
+    },
+    delete: (url: string, config: Object) => {
+        return axiosInstance.delete(url, config);
     },
```

Comment:
Takodje, sad mi sam to primetio, nazivi funkcija nisu konzistentni.
Mislim da bi bilo najbolje da nazivi oslikavaju nazive HTTP metoda.
To znaci da bi trebalo one prve funkcije preimenovati:
- `getResourceList` -> `getOne`
- `getResourceDetails` -> `getList`